fossas/fossa-cli
# Adds license details to attribution

# Overview

Adds license details field to the attribution report.

The existing report did not have information on the copyright info for each license, just the text of the license.  Some customers need to include those copyrights in compliance reports.

The PR adds a new, optional field to our attribution report JSON schema so that the CLI will print it out if it is present.  A new field was chosen over changing the existing field to easily retain backwards compatibility.  The new field is also an object, allowing for further additions.

Previous versions of the CLI _should_ safely ignore this new field.

There is a corresponding core PR to add this field to the API response: https://github.com/fossas/FOSSA/pull/7766

## Acceptance criteria

The CLI supports a field on attribution reports that matches what Core does/will emit containing the copyrights.  This data is passed through to the printed JSON object.

If the field is not present the output should be unaffected.

## Testing plan

1. Ensure that the report is unchanged by running a report against a version of core that does not have the field.
2. Check out the core version that includes the new field.  See https://github.com/fossas/FOSSA/pull/7766
3. Start up core locally from that check-out.
4. Run a report again against that instance.
5. Observe the copyrights in the output.

## References

- [ANE-351](https://fossa.atlassian.net/browse/ANE-351): Add license copyrights to attribution reports

## Checklist
